core/src/main/java/org/arquillian/smart/testing/TestSelection.java

```diff
@@ -14,23 +16,28 @@
 
     private final String className;
 
-    private final Collection<String> types; // TODO or strategy instead (misleading name)
+    private final Collection<String> appliedStrategies;
 
-    public TestSelection(Path location, String... type) {
-        this(new ClassNameExtractor().extractFullyQualifiedName(location), type);
+    private final List<String> testMethodNames;
+
+    public static final TestSelection NOT_MATCHED = new TestSelection("");
+
+    public TestSelection(Path location, String... appliedStrategies) {
+        this(new ClassNameExtractor().extractFullyQualifiedName(location), appliedStrategies);
     }
 
-    public TestSelection(String className, String ... type) {
+    public TestSelection(String className, String ... appliedStrategies) {
         this.className = className;
-        this.types = new LinkedHashSet<>(asList(type));
+        this.appliedStrategies = new LinkedHashSet<>(asList(appliedStrategies));
+        this.testMethodNames = new ArrayList<>();
```

Comment:
Can you point me the place where we are updating this list of `testMethodNames`?

Reply:
https://github.com/arquillian/smart-testing/pull/309/commits/e6314b83a7a4756d23b566994d9f99edc02934d9#diff-b179c0d6894a22602721a54a8efc1d54R49

Reply:
I got confused as it is part of another PR. Thanks

Reply:
Sure, I wanted to keep it separated - the implementation of the core from the code of other strategies, otherwise, it would be one big confusing PR mixing several things together.

Reply:
Again, IDEA is your friend. It's used in the tests. https://github.com/arquillian/smart-testing/blob/468c81b33b72f78eabe053b9f6fb6d68bb3916eb/surefire-provider/src/test/java/org/arquillian/smart/testing/surefire/provider/SmartTestingProviderTest.java#L142

Reply:
that's correct, but I didn't found real logic to filter testMethods

Reply:
I missed followup conversation in the meantime while reviewing PRs, so my comment can be ignored :)